Dev ops environment2025 (#4278)
SkylineCommunications/dataminer-docs: develop/CICD/Skyline DataMiner Software Development Kit/skyline_dataminer_sdk.md

- [**DataMiner Package Project**](xref:skyline_dataminer_sdk_dataminer_package_project)
- **DataMiner Ad Hoc Data Source Project**
- **DataMiner Automation Script Library Project**
- **DataMiner Automation Script Project**
- **DataMiner User-Defined API Project**

## Getting started

To get started, you can:

- **Install the DIS (DataMiner Integration Studio) extension** on Visual Studio, which includes the templates by default.
- **Install the templates directly** using the .NET CLI:

  ```bash
  dotnet new install Skyline.DataMiner.VisualStudioTemplates
  ```

These templates include a helpful wizard for Visual Studio and well-documented CLI arguments for creating projects using the `dotnet` command.

When you create a project, a **GettingStarted.md** file will be added to your project directory. This file provides detailed instructions on how to work with each project type.

## Key features and actions

The Skyline DataMiner SDK hooks into the compilation and publishing processes to streamline the creation of DataMiner application packages and Catalog uploads.

### Compilation: creating DataMiner application packages

> During the compilation of a **Skyline.DataMiner.SDK** project, the SDK checks whether the project property `GenerateDataMinerPackage` is set to `true`. If enabled, it uses the `DataMinerType` property to generate additional output files specific to the project type:

Reviewer: ```suggestion
During the compilation of a **Skyline.DataMiner.SDK** project, the SDK checks whether the project property `GenerateDataMinerPackage` is set to `true`. If this is the case, it uses the `DataMinerType` property to generate additional output files specific to the project type:
```

- **\*.dmapp file**: Represents a DataMiner application package containing the content of your project, ready to be installed on a DataMiner System.
- **\*.zip file**: Represents Catalog information used for publishing the package to the DataMiner Catalog.

### Publishing: uploading to the DataMiner Catalog

> When you publish a **Skyline.DataMiner.SDK** project, the SDK checks whether the `GenerateDataMinerPackage` property is set to `true`. If enabled, it considers the `CatalogPublishKeyName` property, which specifies the name of an environment variable or Visual Studio secret containing a [system key](xref:Managing_DCP_keys#system-keys) or [organization key](xref:Managing_DCP_keys#organization-keys).

Reviewer: ```suggestion
When you publish a **Skyline.DataMiner.SDK** project, the SDK checks whether the `GenerateDataMinerPackage` property is set to `true`. If this is the case, it checks the `CatalogPublishKeyName` property for the name of an environment variable or Visual Studio secret containing a [system key](xref:Managing_DCP_keys#system-keys) or [organization key](xref:Managing_DCP_keys#organization-keys).
```
